### PR TITLE
fix(astro): sub folders not working on Windows

### PR DESCRIPTION
**Pull request by AriPerkkio**
- Fixes https://github.com/stackblitz/tutorialkit/issues/209

### CODE REVIEW

File: packages/astro/src/default/utils/content/files-ref.spec.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  expect(files).toMatchInlineSnapshot(`
+    [
+      "test-fixtures-files.json",
+      [
+        "/first.js",
+        "/nested/directory/second.ts",
+      ],
+    ]
+  `);
+});
```
Comment (AriPerkkio) [inline packages/astro/src/default/utils/content/files-ref.spec.ts:17]: Failure on Windows CI before fix:

>  ❯ src/default/utils/content/files-ref.spec.ts  (1 test | 1 failed) 26ms
>    ❯ src/default/utils/content/files-ref.spec.ts > getFilesRefList returns files
>      → Snapshot `getFilesRefList returns files 1` mismatched
> ⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯
>  FAIL  src/default/utils/content/files-ref.spec.ts > getFilesRefList returns files
> Error: Snapshot `getFilesRefList returns files 1` mismatched
> - Expected
> + Received
>   [
>     "test-fixtures-files.json",
>     [
>       "/first.js",
> -     "/nested/directory/second.ts",
> +     "/nested\directory\second.ts",
>     ],
>   ]

https://github.com/stackblitz/tutorialkit/actions/runs/10287876681/job/28471979082?pr=225

File: packages/cli/tests/create-tutorial.test.ts [hunk 2]
```diff
@@ -67,6 +68,20 @@ test('create and build a project', async (context) => {
...
+  const templateJsons = distFiles.filter((file) => file.startsWith('template-') && file.endsWith('.json'));
+
+  const contents = [...lessonJsons, ...solutionJsons, ...templateJsons].reduce((jsons, current) => {
+    const fileJson = JSON.parse(readFileSync(path.join(dest, 'dist', current), 'utf8'));
+    const filenames = Object.keys(fileJson);
+
+    return { ...jsons, [current]: filenames };
+  }, {});
+
+  expect(JSON.stringify(contents, null, 2)).toMatchSnapshot('built project file references');
```
Comment (AriPerkkio) [inline packages/cli/tests/create-tutorial.test.ts:84]: This assertion is unrelated to the bug. I first thought the problem would be here and created this test case to see if it passes on Windows - it did. 

It seems useful so let's include it as well.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -5,6 +5,7 @@ on:
     branches:
       - main
   pull_request:
+  workflow_dispatch:
```
Comment (AriPerkkio) [inline .github/workflows/ci.yaml:8]: This is useful for contributors. It allows them (and me) to run CI manually on specific branch on forks.

Comment (Nemikolh) [reply .github/workflows/ci.yaml:8]: Oh that's a really nice addition! 😃

**Review by Nemikolh — APPROVED** [review]
Really good find! I like the clean up and the new test! 🤌
